packages/ui/src/components/SelectInputV2/Dropdown.tsx

```diff
@@ -0,0 +1,896 @@
+  refSelect: RefObject<HTMLDivElement>
+  selectedValues: OptionType[]
+  setSelectedValues: Dispatch<SetStateAction<OptionType[]>>
+  isDropdownVisible: boolean
+  setIsDropdownVisible: Dispatch<SetStateAction<boolean>>
+  loadMore?: ReactNode
+  optionalInfoPlacement: 'left' | 'right'
+  isLoading?: boolean
+  selectAll?: { label: ReactNode; description?: string }
+  allSelected: boolean
+  setAllSelected: Dispatch<SetStateAction<boolean>>
+  selectAllGroup?: boolean
+  selectedGroups: string[]
+  setSelectedGroups: Dispatch<SetStateAction<string[]>>
+  numberOfOptions: number
+}
+
+const NON_SEARCHABLE_KEYS = [
+  'Tab',
+  ' ',
+  'Enter',
+  'CapsLock',
+  'Shift',
+  'ArrowDown',
+  'ArrowUp',
+  'ArrowLeft',
+  'ArrowRight',
+  'Escape',
+]
+
+const StyledInfo = styled.div`
+  align-content: center;
+`
+const StyledPopup = styled(Popup)`
+  width: 100%;
+  background-color: ${({ theme }) => theme.colors.neutral.background};
+  color: ${({ theme }) => theme.colors.neutral.text};
+  box-shadow: ${({ theme }) => theme.shadows.dropdown};
+  padding: ${({ theme }) => theme.space[0]};
+`
+
+const DropdownContainer = styled(Stack)`
+  max-height: 256px;
+  overflow-y: scroll;
+  padding: ${({ theme }) => theme.space[0]};
+`
+const DropdownGroup = styled(Stack, {
+  shouldForwardProp: prop => !['selectAllGroup'].includes(prop),
+})<{ selectAllGroup?: boolean }>`
+  background-color: ${({ theme }) => theme.colors.neutral.backgroundWeak};
+  position: sticky;
+  top: 0px;
+  padding-right: ${({ theme }) => theme.space[2]};
+  padding-left: ${({ theme, selectAllGroup }) =>
+    selectAllGroup ? '20px' : theme.space[2]};
+  height: ${({ theme }) => theme.space[4]};
+  text-align: left;
+`
+const DropdownItem = styled('button', {
+  shouldForwardProp: prop => !['disabled', 'selected'].includes(prop),
+})<{
+  disabled: boolean
+  selected: boolean
+}>`
+  text-align:left;
+  border: none;
+  background-color: ${({ theme, selected, disabled }) => {
+    if (selected) {
+      return `${theme.colors.primary.background};`
+    }
+    if (disabled) {
+      return `${theme.colors.neutral.backgroundDisabled};`
+    }
+
+    return `${theme.colors.neutral.background};`
+  }}
+  padding: ${({ theme }) => theme.space['1.5']}
+    ${({ theme }) => theme.space['2']} ${({ theme }) => theme.space['1.5']}
+    ${({ theme }) => theme.space['2']};
+  margin-left: ${({ theme }) => theme.space['0.5']};
+  margin-right: ${({ theme }) => theme.space['0.5']};
+  color:  ${({ theme, disabled }) => (disabled ? `${theme.colors.neutral.textDisabled}` : `${theme.colors.neutral.text}`)};
+
+  &:hover, :focus {
+    background-color: ${({ theme, disabled }) => (disabled ? `${theme.colors.neutral.backgroundStrongDisabled};` : `${theme.colors.primary.background};`)}
+    color: ${({ theme, disabled }) => (disabled ? `${theme.colors.neutral.textStrongDisabled}` : `${theme.colors.primary.text}`)};
+    cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
```

Comment:
Disabled state shows a pointer on the input there is something wrong, it should show `not-allows` instead